### PR TITLE
Use values service in configmap resource

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#6639

Follow up to #115 that wires the code that was moved to the values service.

### CODE REVIEW

File: service/controller/app/v1/resource/configmap/desired_test.go [hunk 1]
```diff
@@ -95,274 +97,12 @@ func Test_Resource_GetDesiredState(t *testing.T) {
 				},
 			},
 		},
-		{
```
Comment (rossf7) [inline service/controller/app/v1/resource/configmap/desired_test.go]: I removed these tests. They are now in the values service.

File: service/controller/app/v1/values/values.go [hunk 2]
```diff
@@ -40,17 +40,6 @@ func New(config Config) (*Values, error) {
...
-// MergeConfigMapData merges configmap data into a single block of YAML that
-// is stored in the configmap associated with the relevant chart CR.
-func MergeConfigMapData(destMap, srcMap map[string]string) (map[string]string, error) {
-	result, err := mergeData(toByteSliceMap(destMap), toByteSliceMap(srcMap))
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
-
-	return toStringMap(result), nil
-}
```
Comment (rossf7) [inline service/controller/app/v1/values/values.go]: This function goes and is replaced by the MergeConfigMapData method.

**Comment by rossf7** [comment]
Tested on `ginger`.

File: service/controller/app/v1/resource/configmap/create_test.go [hunk 3]
```diff
@@ -7,6 +7,7 @@ import (
 
 	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
 	"github.com/giantswarm/apiextensions/pkg/clientset/versioned/fake"
+	"github.com/giantswarm/app-operator/service/controller/app/v1/values"
```
Comment (tomahawk28) [inline service/controller/app/v1/resource/configmap/create_test.go]: gotcha

File: service/controller/app/v1/resource/configmap/delete_test.go [hunk 4]
```diff
@@ -7,6 +7,7 @@ import (
 
 	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
 	"github.com/giantswarm/apiextensions/pkg/clientset/versioned/fake"
+	"github.com/giantswarm/app-operator/service/controller/app/v1/values"
```
Comment (tomahawk28) [inline service/controller/app/v1/resource/configmap/delete_test.go]: gotcha (2)

File: service/controller/app/v1/resource/configmap/desired.go [hunk 5]
```diff
@@ -28,47 +23,14 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
...
-	// We get the app level values if configured.
-	appData, err := r.getConfigMapForApp(ctx, cr)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
-
-	// Config is merged and in case of intersecting values the app level
-	// config is preferred.
-	mergedData, err := values.MergeConfigMapData(catalogData, appData)
+	mergedData, err := r.values.MergeConfigMapData(ctx, cr, cc.AppCatalog)
```
Comment (tomahawk28) [inline service/controller/app/v1/resource/configmap/desired.go]: It gets simpler! <3

File: service/controller/app/v1/resource/configmap/resource.go [hunk 6]
```diff
@@ -4,10 +4,10 @@ import (
 	"reflect"
 
 	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/app-operator/service/controller/app/v1/values"
```
Comment (tomahawk28) [inline service/controller/app/v1/resource/configmap/resource.go]: gotcha (3)

File: service/controller/app/v1/resource/configmap/update_test.go [hunk 7]
```diff
@@ -7,6 +7,7 @@ import (
 
 	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
 	"github.com/giantswarm/apiextensions/pkg/clientset/versioned/fake"
+	"github.com/giantswarm/app-operator/service/controller/app/v1/values"
```
Comment (tomahawk28) [inline service/controller/app/v1/resource/configmap/update_test.go]: gotcha (4)

**Pull request merged**